verilog/analysis/checkers/defparam_rule.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017-2019 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/defparam_rule.h"
+
+#include <string>
+#include <vector>
+
+#include "absl/strings/str_cat.h"
+#include "common/analysis/citation.h"
+#include "common/analysis/lint_rule_status.h"
+#include "common/analysis/matcher/bound_symbol_manager.h"
+#include "common/analysis/matcher/matcher.h"
+#include "common/text/symbol.h"
+#include "common/text/syntax_tree_context.h"
+#include "verilog/analysis/descriptions.h"
+#include "verilog/analysis/lint_rule_registry.h"
+
+namespace verilog {
+namespace analysis {
+
+using verible::GetStyleGuideCitation;
+
+// Register the lint rule
+VERILOG_REGISTER_LINT_RULE(DefparamRule);
+
+absl::string_view DefparamRule::Name() { return "defparam"; }
+const char DefparamRule::kTopic[] = "defparam";
+const char DefparamRule::kMessage[] =
+    "Do not use defparam.";
+
+std::string DefparamRule::GetDescription(
+    DescriptionType description_type) {
+  return absl::StrCat(
+      "Do not use defparam. See:",
+      GetStyleGuideCitation(kTopic), ".");
+}
+
+void DefparamRule::HandleSymbol(
+    const verible::Symbol& symbol, const verible::SyntaxTreeContext& context) {
+  verible::matcher::BoundSymbolManager manager;
+  if (matcher_.Matches(symbol, &manager)) {
+    violations_.push_back(verible::LintViolation(symbol, kMessage, context));
```

Comment:
I recommend narrowing the range to that of the 'defparam' token itself, which should be in child position [0].

Reply:
Thank you for the review!

Could you please elaborate on that? As far as I can see some other checkers try to use the BoundMap from the manager to extract more detailed info, but it is empty in this case. Whose child node should be used here, should it be extracted from the context? And what is the proper way to access it?

Reply:
`violations_.push_back(verible::LintViolation( *(context.top().children()[0].get()), kMessage, context));`

Seems to work, is that what you meant?

Reply:
Changed the implementation to what I pasted above (+ some sanity checks), please let me know if that's what you had in mind